src/components/Posts/PostItem.tsx

```diff
@@ -31,14 +33,14 @@ type PostProps = {
   post: Post;
 };
 
-const likePost = async (userId: number, postId: number) => {
+const likePost = async (userId: number | undefined, postId: number) => {
```

Comment:
It was working fine before since I hard coded this to be 1, but we shouldn't allow an undefined type here since the server will throw an error if undefined actually got passed

We either need to do an undefined check or make it so that the `userId` property on the `user` object can't be undefined. I'm inclined to do the second option because really a `user` object shouldn't be able to exist without a `userId`

Reply:
We'll create a bigger ticket around general error handling and tackle this there. An idea was to do undefined checks higher up in the component tree and redirect users to another route if their `userId` is undefined